src/Migrations/M032MigrateSocialMediaTwitterBlockToEmbedBlock.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+// phpcs:disable Generic.Files.LineLength.MaxExceeded
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+use WP_Block_Parser;
+
+/**
+ * Migrate Social Media blocks with Twitter/X urls to Embed blocks.
+ */
+class M032MigrateSocialMediaTwitterBlockToEmbedBlock extends MigrationScript
+{
+    private const SOURCE_INSTAGRAM = 'instagram';
+    private const SOURCE_FACEBOOK = 'facebook';
+    private const SOURCE_TWITTER = 'twitter';
+    private const SOURCE_X = 'x';
+
+    private const SOURCE_PATTERNS = [
+        self::SOURCE_INSTAGRAM => '/^(https?\:\/\/)?(www\.)?instagram\.com\/.+$/i',
+        self::SOURCE_FACEBOOK => '/^(https?\:\/\/)?(www\.)?facebook\.com\/.+$/i',
+        self::SOURCE_TWITTER => '/^(https?\:\/\/)?(www\.)?(twitter\.com)\/.+$/i',
+        self::SOURCE_X => '/^(https?\:\/\/)?(www\.)?x\.com\/.+$/i',
+    ];
+
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    protected static function execute(MigrationRecord $record): void
+    {
+        try {
+            // Get the list of posts using Media blocks.
+            $posts = Utils\Functions::get_posts_using_specific_block(
```

Comment:
This function has to be adjusted to include all post statuses.
Check it out here: https://github.com/greenpeace/planet4-master-theme/blob/a98281c2d697f301c6aa32bdecabde76e69b1e56/src/Migrations/M033MigrateCampaignCoversToP4ColumnsBlock.php#L26